### DISCUSSION TITLE
Possible to import Dialogic Timelines from other VN scripts?

### DISCUSSION THREAD

**Discussion by McStebb** [dialogic-godot/dialogic#1764] [Q&A] [answered]
Hello,

I am one of the many refugees swept up by the idiotic direction that Unity is headed in.  I have a visual novel project that my team made a decent start on in Unity (via Naninovel, specifically) and we'd like to port it over to Godot/Dialogic

My question is this...

How can I import timelines that were edited outside of Dialogic and get Dialogic to recognize the changes?  It seems like if I make a change to a Dialogic json file and then try to view the changes in-engine, Dialogic seems to erase my changes and only seems to preserve changes if I make them using the in-engine editor.

My (unobtainium) end goal is to write a script, probably in PowerShell or something, that will take our Naninovel scripts as input and then output the closest equivalent Dialogic json timeline... but the fact that I can't edit the Dialogic timelines outside of the engine is a pretty big roadblock to that plan.

The alternative, of course, is to go line by line and recreate the VN by hand... but I'd prefer not to have to resort to that.

Let me know if you folks have any ideas.
Thanks!

**Comment by Jowan-Spooner** [comment, marked as answer]
Hey, sorry to hear you've been affected by the recent unity stuff.

This largely depends on whether you are using dialogic 1 or dialogic 2.

In v1 timelines are indeed stored as jsons. I think it should be possible to edit them, but haven't used v1 for quite some time. Only thing I can suggest is not having the editor open while you make the changes.

In v2 dialogic uses plain text files with the  .dtl extension for recognizing them. These use a custom syntax and I'm pretty sure they can be created outside and edited outside too.

**Comment by McStebb** [comment]
After examining the two options it seems like Dialogic 2 is far and away going to be my best bet. The syntax is close _enough_ to Naninovel that I should be able to automate at least the majority of the port.

Thanks for pointing me in the right direction.
